pkg/transferer/load.go

```diff
@@ -0,0 +1,164 @@
+package transferer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	"github.com/valyala/fasthttp"
+	"net/http"
+	"strconv"
+	"sync"
+	"time"
+)
+
+type LoadStatus int
+
+const (
+	LoadStatusNone LoadStatus = iota
+	LoadStatusOK
+	LoadStatusWait
+	LoadStatusTerminate
+)
+
+type LoadChecker struct {
+	c             *fasthttp.Client
+	connectionURL string
+
+	thresholds []Threshold
+
+	m            sync.RWMutex
+	latestStatus LoadStatus
+}
+
+func NewLoadChecker(ctx context.Context, c *fasthttp.Client, url string) *LoadChecker {
+	thresholds := []Threshold{
+		{
+			Key:          "cpu",
+			Query:        `100 - (avg by (instance) (rate(node_cpu_seconds_total{mode="idle",node_name="pmm-server"}[1m])) * 100)`,
+			MaxLoad:      50,
+			CriticalLoad: 70,
+		},
+	}
+	lc := &LoadChecker{
+		c:             c,
+		connectionURL: url,
+		thresholds:    thresholds,
+		latestStatus:  LoadStatusNone,
+	}
+	lc.runStatusUpdate(ctx)
+	return lc
+}
+
+func (c *LoadChecker) GetLatestStatus() LoadStatus {
+	c.m.RLock()
+	defer c.m.RUnlock()
+	return c.latestStatus
+}
+
+func (c *LoadChecker) setLatestStatus(s LoadStatus) {
+	c.m.Lock()
+	defer c.m.Unlock()
+	c.latestStatus = s
+}
+
+func (c *LoadChecker) runStatusUpdate(ctx context.Context) {
+	go func() {
+		ticker := time.NewTicker(time.Second) // TODO: make duration configurable
+		defer ticker.Stop()
+		for range ticker.C {
```

Comment:
`ticker.C` is also a channel so better to put it into a select statement

also, please, add logs (new goroutine started, new loop, finishing, etc.) as in the other concurrent code blocks

Reply:
resolved in eead670702e64ee87b033ffcb2a6167bb31de41b